UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs

```diff
@@ -1,51 +1,80 @@
         private readonly ILogger<FulfilmentDataService> logger;
+        private readonly IOptions<FileShareServiceConfiguration> fileShareServiceConfig;
+        private readonly IConfiguration configuration;
 
-        public FulfilmentDataService(ISalesCatalogueStorageService scsStorageService, IAzureBlobStorageService azureBlobStorageService, 
+        public FulfilmentDataService(IAzureBlobStorageService azureBlobStorageService, 
                                     IFulfilmentFileShareService fulfilmentFileShareService,
-                                    IOptions<EssFulfilmentStorageConfiguration> storageConfig, ILogger<FulfilmentDataService> logger)
+                                    ILogger<FulfilmentDataService> logger,
+                                    IOptions<FileShareServiceConfiguration> fileShareServiceConfig,
+                                    IConfiguration configuration)
         {
-            this.scsStorageService = scsStorageService;
             this.azureBlobStorageService = azureBlobStorageService;
             this.fulfilmentFileShareService = fulfilmentFileShareService;
-            this.storageConfig = storageConfig;
             this.logger = logger;
+            this.fileShareServiceConfig = fileShareServiceConfig;
+            this.configuration = configuration;
         }
 
         public async Task<string> CreateExchangeSet(SalesCatalogueServiceResponseQueueMessage message)
-        {            
-            var fssFileName = $"{message.BatchId}-fssresponse.json";
-
-            string storageAccountConnectionString = scsStorageService.GetStorageAccountConnectionString();
+        {
+            string homeDirectoryPath = configuration["HOME"];
+            var exchangeSetRootPath = Path.Combine(homeDirectoryPath, DateTime.UtcNow.ToString("ddMMMyyyy"), message.BatchId, fileShareServiceConfig.Value.ExchangeSetFileFolder, fileShareServiceConfig.Value.EncRoot);
 
-            var response = await azureBlobStorageService.DownloadSalesCatalogueResponse(message.ScsResponseUri);
+            var response = await DownloadSalesCatalogueResponse(message);
             if (response.Products != null && response.Products.Any())
             {
-                logger.LogInformation(EventIds.QueryFileShareServiceRequestStart.ToEventId(), "Query File share service request started for {BatchId}", message.BatchId);
-                var searchBatchResponse = await fulfilmentFileShareService.QueryFileShareServiceData(response?.Products);
-                logger.LogInformation(EventIds.QueryFileShareServiceRequestCompleted.ToEventId(), "Query File share service request completed for {BatchId}", message.BatchId);
+                int allowedThreadPerEs = fileShareServiceConfig.Value.ParallelSearchTaskCount;
+                int pgGroupCount = response.Products.Count % allowedThreadPerEs == 0 ? response.Products.Count / allowedThreadPerEs : (response.Products.Count / allowedThreadPerEs) + 1;
```

Comment:
Kindly rename variables allowedThreadPerEs to parallelSearchTaskCount and pgGroupCount to productGroupCount

Reply:
updated variables name